allihoopa-sdk/src/main/java/com/allihoopa/sdk/AllihoopaPiece.java

```
@@ -236,4 +257,28 @@ public void getUrlWithFormat(String format, AudioWithFormatCallback callback) {
         void onComplete(@Nullable String url);
         void onError(Exception e);
     }
+
+    public static final class Attachment {
+        public URL attachmentURL;
+        public String attachmentMimeType;
+
+        public Attachment(URL attachmentURL, String attachmentMimeType) {
+            attachmentURL = attachmentURL;
+            attachmentMimeType = attachmentMimeType;
+        }
+
+        public static Attachment fromJSON(JSONObject attachmentObj, String attachmentMimeType) {
+            if (attachmentObj == null) {
```

I would much rather this throw an exception if `attachmentObj` is `null` (using e.g. `Objects.requireNonNull`). Silently accepting `null` could lead to hard-to-find bugs later on